include/nbl/builtin/hlsl/emulated_float64_t.hlsl

```diff
@@ -0,0 +1,317 @@
+#include <nbl/builtin/hlsl/cpp_compat.hlsl>
+
+using float32_t = float;
+//using emulated_float64_t = double;
+
+namespace emulated
+{
+    namespace impl
+    {
+        nbl::hlsl::uint32_t2 umulExtended(uint32_t lhs, uint32_t rhs)
+        {
+            uint64_t product = uint64_t(lhs) * uint64_t(rhs);
+            nbl::hlsl::uint32_t2 output;
+            output.x = (product & 0xFFFFFFFF00000000) >> 32;
+            output.y = product & 0x00000000FFFFFFFFull;
+            return output;
+        }
+
+        nbl::hlsl::uint32_t2 add64(uint32_t a0, uint32_t a1, uint32_t b0, uint32_t b1)
+        {
+           nbl::hlsl::uint32_t2 output;
+           output.y = a1 + b1;
+           output.x = a0 + b0 + uint32_t(output.y < a1);
+
+           return output;
+        }

```

Comment:
replace usages and occurences just with `a+b` we require 64bit integer math

Reply:
same goes for sub

Reply:
cannot remove from `mul64to128`

Reply:
need to rewrite `mul64to128` , see https://github.com/Devsh-Graphics-Programming/Nabla/pull/696#discussion_r1659881243

Reply:
irrevelant